fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/TransactionImpl.java

```diff
@@ -278,6 +283,35 @@ public void refresh() {
     @Override
     public void lockResource(final FedoraId resourceId) {
         getResourceLockManger().acquire(getId(), resourceId);
+        lockGhostNodesOfResource(resourceId);
```

Comment:
I was unsure if these two should be wrapped in a `try { } catch` for a `ConcurrentUpdateException` to release all the locks if one fails 🤷